homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -1030,19 +1017,19 @@ def __init__(self, name, device, entry, unique_id):
         if self._model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
             self._device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB
             self._available_attributes = AVAILABLE_ATTRIBUTES_AIRHUMIDIFIER_CA_AND_CB
-            self._speed_list = [
+            self._preset_modes = [
                 mode.name
                 for mode in AirhumidifierOperationMode
                 if mode is not AirhumidifierOperationMode.Strong
             ]
         elif self._model in [MODEL_AIRHUMIDIFIER_CA4]:
             self._device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA4
             self._available_attributes = AVAILABLE_ATTRIBUTES_AIRHUMIDIFIER_CA4
-            self._speed_list = [SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
```

Comment:
These actually look like speeds.  These should be implemented as percentages in the new model.

Reply:
It was implemented as speed in the past to get the device support merged. The operation mode "auto" is missing here. I would like to stick to presets.

Reply:
Or should I implement both:

```
set_speed ['low', 'medium', 'high']
set_preset_mode ['auto', 'low', 'medium', 'high']
```

To be Alexa and Google Home compatible?

Reply:
Preset mode should map to auto, manual (or none), silent (or night) and levels. Fan speed in percentage is possible in manual mode (used to be favorite levels now is motor speed). Levels is low/medium/high or 1/2/3.

Reply:
For this one keep `self._speed_list`, and if @mouth4war if confirming the device also supports a list of preset modes [auto, manual (or none), silent (or night) and levels] then set `self._preset_modes` as well

```python
from homeassistant.util.percentage import ordered_list_item_to_percentage
```
```python

    @property
    def percentage(self) -> Optional[int]:
        """Return the current speed percentage."""
        if not self._speed_list:
            return None
        return ordered_list_item_to_percentage(self._speed_list, current_speed)

    @property
    def speed_count(self) -> int:
        """Return the number of speeds the fan supports."""
        if not self._speed_list:
            return None
        return len(self._speed_list)
```

Reply:
It seems I made the comment on the wrong PR. The comment was meant for miio air purifier fan. Apologies.